tiledb/group.py

```diff
@@ -264,6 +265,11 @@ def __init__(
         config: Config = None,
         ctx: Optional[Ctx] = None,
     ):
+        if uri is not None and tiledb.object_type(uri) != "group":
```

Comment:
Can you clarify why we need this - the `tiledb::Group` constructor should throw if the uri is not a group?

We should avoid extra calls to introspect like this, if we can (they are expensive on remote/object stores).

Reply:
If we need this, it seems like we are missing something either in the wrapping or in core error handling/propagation.

Reply:
I agree 100%.